engines/sci/graphics/controls16.cpp

```diff
@@ -328,7 +328,11 @@ void GfxControls16::kernelDrawButton(Common::Rect rect, reg_t obj, const char *t
 		_paint16->frameRect(rect);
 		rect.grow(-2);
 		_ports->textGreyedOutput(!(style & SCI_CONTROLS_STYLE_ENABLED));
-		_text16->Box(text, languageSplitter, false, rect, SCI_TEXT16_ALIGNMENT_CENTER, fontId);
+		if (!g_sci->hasMacFonts()) {
+			_text16->Box(text, languageSplitter, false, rect, SCI_TEXT16_ALIGNMENT_CENTER, fontId);
+		} else {
+			_text16->macDraw(text, rect, SCI_TEXT16_ALIGNMENT_CENTER, fontId, _text16->GetFontId(), 0);
+		}
```

Comment:
Perhaps it would make sense to call `macDraw()` from within `Box()`, and keep the calling code a bit cleaner?

Reply:
I think this structure is *just slightly* better because...
- It matches the function structure in the SSCI executables. They forked the internal "R__" functions, added "Mac" to their names, and called them here.
- We already have to do it this way for kTextSize since it can call either the Mac or non-Mac `GfxText16` functions depending on an extra parameter. I think it's better to be consistent and call both of the `GfxText16` mac functions conditionally from kernel code.
- `GfxControls16::kernelDrawButton` introduces a mac-only text drawing step for inversion. I think it's better to call the mac text function explicitly instead of adding a fake reference to `Box`. `Box` is the kind of important function i do find-references on all the time, and this would be the type of reference you'd have to figure out each time that it doesn't really apply when doing real (non-mac) work.
- `Box` is huge and important, I'd rather not add even more to it, especially since this Mac stuff is an obscure feature.
- It makes it clear that the language splitter doesn't apply to Mac versions (a `Box` parameter).
- Now that I've typed all this, if I compromise I will show weakness [ just kidding! ]

Reply:
Haha. No compromises then!
I see your point. Makes sense to keep the structure as-is